### DISCUSSION TITLE
What's the right way to make Bytes (de)serializeable?

### DISCUSSION THREAD

**Discussion by thor314** [rwf2/Rocket#2328] [Questions] [answered]
I would like to have a message that passes around some raw bytes:
```rust
#[derive(Debug, Clone, Serialize, Deserialize)]
#[cfg_attr(test, derive(PartialEq, Eq))]
#[serde(crate = "rocket::serde")]
pub struct SigningMessage {
  pub payload: rocket::http::hyper::body::Bytes,
}
```

https://docs.rs/rocket/0.5.0-rc.2/rocket/http/hyper/body/struct.Bytes.html

But I see that `Bytes` isn't `Serialize`. I could write a wrapper of Bytes that I implement `Serialize` for, but I wanted to ask if there was a good reason it wasn't already.

**Comment by edgarogh** [comment, marked as answer]
`rocket::http::hyper::body::Bytes` is just a reexport of [`bytes::Bytes`](https://docs.rs/bytes/latest/bytes/struct.Bytes.html), from a completely independent crate that's very popular in the Rust community.

If you look at [its feature flags](https://docs.rs/crate/bytes/latest/features), you'll see that it does support `serde`. You just have to explicitly enable it yourself by depending on `::bytes` :

```toml
[dependencies]
...
bytes = { version = "1.0", default-features = false, features = ["serde"] }
```
